src/pages/Edit/Indicent.tsx

```diff
@@ -0,0 +1,169 @@
+      }
+    },
+  });
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setValue,
+    watch,
+    reset,
+  } = useForm({
+    resolver: zodResolver(schema as ZodTypeAny),
+  });
+
+  const currentUserQuery = useQueryWithClient("currentUser", async (client) =>
+    getCurrentUser(client)
+  );
+
+  const submitMutation = useQueryMutationWithClient<Incident>((client, data) =>
+    editIncident(
+      client,
+      incidentId as string,
+      data,
+      currentUserQuery.data?.user?.email as string
+    )
+  );
+
+  const incidentMeta = useQueryWithClient(["incidentMeta"], (client) =>
+    getCreateIncidentMeta(client)
+  );
+
+  const incidentByIdQuery = useQueryWithClient(
+    "getIncidentById",
+    (client) => getIncidentById(client, incidentId as string),
+    {
+      enabled: !!incidentId,
+    }
+  );
+
+  useEffect(() => {
+    if (!submitMutation.isSuccess) return;
+
+    navigate("/redirect");
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [submitMutation.isSuccess]);
+
+  useEffect(() => {
+    if (inputs.length === 0 || !incidentMeta.isSuccess) return;
+
+    const newObj: { [key: string]: ZodTypeAny } = {};
+
+    inputs.forEach((field) => {
+      if (field.required) {
+        newObj[field.name] = z.string().nonempty();
+      } else if (field.name === "assignments") {
+        newObj[field.name] = z.array(z.string()).optional();
+      } else {
+        newObj[field.name] = z.string().optional();
+      }
+    });
+    setSchema(getIncidentSchema(inputs, newObj, incidentMeta.data));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [incidentMeta.isSuccess]);
+
+  useEffect(() => {
+    if (!incidentByIdQuery.data) return;
+
+    const incident = incidentByIdQuery.data.incident;
+    const data = {
+      ...incident,
+      assignments: incident.assignments?.map((a) => a.assignee.id),
+      services: incident.service.id,
+      priority: incident.priority?.id,
+    };
+
+    reset(data);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [incidentByIdQuery.isSuccess]);
+  if (!incidentMeta.isSuccess || !incidentByIdQuery.isSuccess)
+    return <LoadingSpinnerCenter />;
+
+  return (
+    <form
+      onSubmit={handleSubmit((data) => {
+        data.incident.type = "incident_reference";
+        submitMutation.mutate(data);
+      })}
+      style={{ width: "100%" }}
+    >
+      <Stack vertical style={{ width: "100%" }} gap={6}>
+        {inputs.map((field, i) => (
+          <FieldMappingInput
+            key={i}
+            dropdownData={incidentMeta.data}
+            errors={errors}
+            field={field}
+            register={register}
+            setValue={setValue}
+            watch={watch}
+          />
+        ))}
+        <Stack justify="space-between" style={{ width: "100%" }}>
+          <Button
+            type="submit"
+            text={submitMutation.isIdle ? "Edit" : "Editing..."}
```

Comment:
here you can not change the text, but add prop `loading={submitMutation.isIdle}`. It will be more user-friendly

Reply:
Please make changes in all forms